clientv3/logger.go

```diff
@@ -35,19 +38,31 @@ type settableLogger struct {
 
 func init() {
 	// use go's standard logger by default like grpc
-	logger.mu.Lock()
-	logger.l = log.New(os.Stderr, "", log.LstdFlags)
-	grpclog.SetLogger(&logger)
-	logger.mu.Unlock()
+	logger.set(log.New(os.Stderr, "", log.LstdFlags))
+
+	// disable client side grpc by default
+	grpclog.SetLogger(log.New(ioutil.Discard, "", 0))
```

Comment:
this is racey iirc, that's why this logger thing exists in the first place-- the logger has to override the `grpclog` at initialization so that any changes to the grpclog go through `logger` with locking instead of through `SetLogger`; it should be `grpclog.SetLogger(&logger)` then only updates through `logger.set(l)`.

```go
logger.l = log.New(io.util.Discard, "", 0))
grpclog.SetLogger(&logger)
```